app.js

```diff
@@ -2,16 +2,37 @@ const express = require("express");
 const app = express();
 const {getAllTopics} = require("./controllers/topics_controllers")
 const {getAllEndpoints} = require("./controllers/endpoints_controllers")
+const {getArticleById} = require("./controllers/articles_controllers")
 
 app.use(express.json());
 
-app.get("/api/topics", getAllTopics)
 app.get("/api",getAllEndpoints)
 
+app.get("/api/topics", getAllTopics)
+
+
+app.get("/api/articles/:article_id", getArticleById)
+
 app.get("*",(req,res,next) => {
     res.status(404).send({ msg: "Route does not exist" })
 })
 
+
+app.use((err, req, res, next) => {
+    if (err.code === "22P02") {
+        res.status(400).send({msg: "Bad request!"});
+    } else {
+        next(err);
+    }
+});
+app.use((err, req, res, next) => {
+    if (err.status) {
+        res.status(err.status).send({msg: err.msg});
+    } else {
+        next(err);
+    }
+});
+
```

Comment:
Excellent error handling here! It might be nice to create a new file to put all these, so they don't take over your app.js :)